airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -433,15 +437,27 @@ public JobInfoRead resetConnection(final ConnectionIdRequestBody connectionIdReq
   }
 
   public ConnectionState getState(final ConnectionIdRequestBody connectionIdRequestBody) throws IOException {
-    final Optional<State> currentState = configRepository.getConnectionState(connectionIdRequestBody.getConnectionId());
-    LOGGER.info("currentState server: {}", currentState);
-
-    final ConnectionState connectionState = new ConnectionState()
-        .connectionId(connectionIdRequestBody.getConnectionId());
+    final UUID connectionId = connectionIdRequestBody.getConnectionId();
+    return getStateFromConnectionUUID(connectionId);
+  }
 
-    currentState.ifPresent(state -> connectionState.state(state.getState()));
+  public ConnectionState updateState(final ConnectionUpdateStateBody connectionUpdateStateBody) throws IOException {
+    final UUID connectionId = connectionUpdateStateBody.getConnectionId();
+    final Set<ConfigType> configTypes = new HashSet();
+    configTypes.add(ConfigType.SYNC);
+    final List<Job> runningJobs = jobPersistence.listJobs(configTypes, connectionId.toString(), 2, 0)
+        .stream().filter(j -> !j.isJobInTerminalState())
+        .collect(Collectors.toList());
+    System.out.println(runningJobs);
+
+    // TODO: Ideally we would throw or return a message to the user here that the update was ignored because a sync was running
```

Comment:
we should add a thing in the docs that says how state can be updated, and makes note of this limitation. Can we return a response that gives the user feedback on the UI about this limitation?

Reply:
I'm sure that it's possible, I just need to sort out how :D